docs/upgrade.md

```diff
@@ -3,5 +3,261 @@ title: Upgrade guide
 description: Guide to update between major Powertools for AWS Lambda (TypeScript) versions
 ---
 
-!!! warning
-    This guide is a work in progress. We'll update it as we get closer to the 2.0 release. If you have any questions, or want to follow the progress, please join the discussion on the [GitHub issue](https://github.com/aws-powertools/powertools-lambda-typescript/issues/1714).
\ No newline at end of file
+## Migrate from v1 to v2
+
+We've made targeted breaking changes to make your transition to v2 as smooth as possible while still providing you with the most up-to-date features and patterns. This guide will help you migrate your existing codebase to v2.
+
+### Quick summary
+
+| Area                                   | Change                                                                                                                                           | Code change required |
+| -------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------ | -------------------- |
+| **ES Modules support**                 | Added ES Modules support via dual CommonJS and ES Module bundling, this enables top-level `await` and tree-shaking.                              | No                   |
+| **Middy.js middleware imports**        | Updated import path for Middy.js middlewares to leverage subpath exports - i.e. `@aws-lambda-powertools/tracer/middleware`.                      | Yes                  |
```

Comment:
This only applies in cases when customers already use middy, and having a general "Yes" in the last column might be not clear. Can we add a note, something like "only if you are using middy".

Reply:
The line item starts with `Middy.js`, I think it's clear enough. As a reader if I'm not using middy or don't know what it is, I'd stop paying attention to that line.

In this case I'd be more inclined to prioritize keeping the table/page as little dense as possible.